tests/unittests/view_tests.py

```diff
@@ -2098,7 +2104,21 @@ def test_exclude_fields(self):
             with self.assertRaises(AttributeError):
                 sample.exclude_fields_field1
 
-            self.assertIsNone(sample.exclude_fields_field2)
+            self.assertEqual(sample.exclude_fields_field2, 1)
```

Comment:
Suggestion: move the `stats()`-specific tests into separately-named test methods? 

This bug is strictly the fault of interplay between `$bsonSize + $project`, which could occur in more than just the `select_fields()` and `exclude_fields()` code paths. This is very minor, of course.

Reply:
Sure thing

Reply:
Done